flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -620,31 +620,36 @@ struct StringLitOpConversion : public FIROpConversion<fir::StringLitOp> {
     if (attr.isa<mlir::StringAttr>()) {
       rewriter.replaceOpWithNewOp<mlir::LLVM::ConstantOp>(constop, ty, attr);
     } else {
-      // FIXME: As of llvm head 85bf221f204eafc1142a064f1650ffa9d9e03dad, using
-      // a dense elements attr causes llvm ir verification error:
-      //   %0 = llvm.mlir.constant(dense<[234, 456]> : vector<2xi16>) :
```

Comment:
Isn't the FIXME still valid ?

A chain of inserts has terrible MLIR compile time performance if we have a giga byte string constant, shouldn't we aim at generating an mlir::DenseAttr here ?

Illustration:
```
module m
  integer, parameter :: n = 100000 
  character(kind=2, len=n) :: c = repeat(2_"abcd", n/4)
end module
```

`time bbc test.f90 -o test.mlir` : `real	0m0.086s`
`time tco test.mlir -o test.ll` : `real	59m50.630s`

Everything went fast until the LLVM dialect to LLVM IR pass that takes nearly 1 hour for this example because of the `mlir::LLVM::InsertValueOp` folding hell.

Reply:
No. This patch fixes that.

Thanks for the heads up about premature folding.